pandas/tests/io/formats/test_to_csv.py

```diff
@@ -15,8 +15,31 @@
 import pandas._testing as tm
 
 
+@pytest.fixture(params=["python", "pyarrow"])
+def engine(request):
+    # TODO: Skip if pyarrow not found
+    if request.param == "pyarrow":
+        pytest.importorskip("pyarrow")
+    return request.param
+
+
+@pytest.fixture
+def pyarrow_xfail(request):
+    """
+    Fixture that xfails a test if the engine is pyarrow.
+    """
+    engine = request.getfixturevalue("engine")
+    if engine == "pyarrow":
+        mark = pytest.mark.xfail(reason="pyarrow doesn't support this.")
+        request.node.add_marker(mark)
+
+
+xfail_pyarrow = pytest.mark.usefixtures("pyarrow_xfail")
+
+
 class TestToCSV:
-    def test_to_csv_with_single_column(self):
+    @xfail_pyarrow
```

Comment:
Like we've been doing with the read_csv test recently, if the test fails due to an exception from pandas and not pyarrow we should check for that exception

Reply:
Updated most of these.

Since some test multiple things in a test, I left the xfail_pyarrow on some since sometimes it would fail on an earlier assert due to mismatch between the pyarrow and python output.